src/pages/MainPage.jsx

```diff
@@ -1,11 +1,28 @@
-import React from "react";
+import React, { useState } from "react";
 import GNBBOX from "../components/common/GNBBOX";
 import MainCarousel from "../components/main/MainCarousel";
+import MainHeaderRow from "../components/main/MainHeaderRow";
+import MainSearchBar from "../components/main/MainSearchBar";
 
 export default function MainPage() {
+  const [isSearchBarOpen, setIsSearchBarOpen] = useState(false);
+
+  const handleOpenSearchBar = () => {
+    setIsSearchBarOpen(true);
+  };
+  const handleCloseSearchBar = () => {
+    setIsSearchBarOpen(false);
+  };
+
   return (
     <div className="flex w-full h-full flex-col">
-      <div className="w-full h-full overflow-x-hidden">
+      <div className="w-full h-full ">
+        {isSearchBarOpen && (
+          <MainSearchBar handleCloseSearchBar={handleCloseSearchBar} />
+        )}
+        {!isSearchBarOpen && (
+          <MainHeaderRow handleOpenSearchBar={handleOpenSearchBar} />
+        )}
```

Comment:
여기는 삼항 연사자가 들어가는 쪽이 보기 더 좋을 것 같습니다!
```suggestion
        {isSearchBarOpen ? (
          <MainSearchBar handleCloseSearchBar={handleCloseSearchBar} />
        ) : (
          <MainHeaderRow handleOpenSearchBar={handleOpenSearchBar} />
        )}
```